packages/transformers/js/core/src/env_replacer.rs

```diff
@@ -41,6 +42,25 @@ impl<'a> Fold for EnvReplacer<'a> {
       }
     }
 
+    if let Expr::Bin(ref binary) = node {
+      if let Expr::Member(ref member) = &*binary.right {
+        if let Expr::Lit(Lit::Str(ref left)) = &*binary.left {
+          if let Expr::Ident(ref ident) = &*member.obj {
+            if !self.decls.contains(&id!(ident)) && &ident.sym == "process" {
+              if let MemberProp::Ident(ref prop) = member.prop {
+                if &prop.sym == "env" {
```

Comment:
Maybe we should use the `match_member_expr` utility to match `process.env` as below?